### PR TITLE
Android: Changed to select raw image quality when using localhost(127.0.0.1)

### LINKED ISSUE #102: Add encoding selector
I want you to support raw image quality because it is used in the local environment with termux etc.

### PR DESCRIPTION
**Pull request by nagasawa**
Please make sure you've read and understood the [contribution guidelines](../CONTRIBUTING.md).

Fixes #102 .

Changes proposed in this pull request:

*Android: Changed to select raw image quality when using localhost(127.0.0.1)

### CODE REVIEW

File: android/app/src/main/cpp/vncconn.c [hunk 1]
```diff
@@ -506,4 +514,4 @@ JNIEXPORT jboolean JNICALL Java_com_coboltforge_dontmind_multivnc_VNCConn_rfbSen
     }
     else
         return JNI_FALSE;
-}
\ No newline at end of file
+}
```
Comment (bk138) [inline android/app/src/main/cpp/vncconn.c:517]: please remove unrelated changes

File: android/app/src/main/cpp/vncconn.c [hunk 2]
```diff
@@ -369,6 +369,14 @@ JNIEXPORT jboolean JNICALL Java_com_coboltforge_dontmind_multivnc_VNCConn_rfbIni
         cl->serverHost = strdup(cHost);
         (*env)->ReleaseStringUTFChars(env, host, cHost);
     }
+    if (0 == strcmp("localhost", cl->serverHost) || 0 == strcmp("127.0.0.1", cl->serverHost))
+    {
+        cl->appData.encodingsString = "raw";
+    }
+    else
```
Comment (bk138) [inline android/app/src/main/cpp/vncconn.c:376]: please leave the non-localhost case to the library's default. i.e. remove the else branch.

File: android/app/src/main/cpp/vncconn.c [hunk 3]
```diff
@@ -369,6 +369,14 @@ JNIEXPORT jboolean JNICALL Java_com_coboltforge_dontmind_multivnc_VNCConn_rfbIni
         cl->serverHost = strdup(cHost);
         (*env)->ReleaseStringUTFChars(env, host, cHost);
     }
+    if (0 == strcmp("localhost", cl->serverHost) || 0 == strcmp("127.0.0.1", cl->serverHost))
```
Comment (bk138) [inline android/app/src/main/cpp/vncconn.c:372]: * style: please put the 0 as second place in the comparison
* logic: please also handle the IPv6 case.

**Review by bk138 — CHANGES_REQUESTED** [review]
Thank you very much! Some changes needed, also please make the commit message (as noted in CONTRIBUTING.md):

* shorter
* have a wording  so the sentence "when applied, this commit will ..." applies

**Comment by bk138** [comment]
@nagasawa any news on this?

**Comment by nagasawa** [comment]
I'm sorry, but I'm working on weekdays and I don't have enough time, so I have to deal with Saturdays and Sundays.

**Comment by nagasawa-umebius** [comment]
dce10b8

I've committed the fix.
I couldn't fix it because I don't know why the change appears on the last line.

**Comment by bk138** [comment]
I've taken the freedom to modify your changes as proposed and use a proper commit message via 208b86a3ed3630edbd8eba12786c92d96eb66079.

**Pull request closed without merging**